File: 卡码/KamaCoder/Js/test10.js
//10. 运营商活动
const rl=require("readline").createInterface({input:process.stdin});
const iter=rl[Symbol.asyncIterator]();
const readline=async ()=>(await iter.next()).value;

async function main(){
  while(line=await readline()){
    let [m,k]=line.split(" ").map(Number);
    if(m==0&&k==0) break;
    let total=m;
    while(m>=k){
      let left=Math.floor(m/k);
      total+=left;
      m=left+m%k;
    }
    console.log(total);
  }
}

main();

File: 卡码/KamaCoder/Python/test10.py
#10. 运营商活动
import sys
def main():
	for line in sys.stdin:
		[m,k]=list(map(int,line.split()))
		if m==0 and k==0:
			break
		total=m
		while m>=k:
			left=m//k
			total+=left
			m=left+m%k
		print(total)

if __name__ == '__main__':
	main()
